### PR TITLE
[data] correct api annotation for tfrecords_datasource

### PR DESCRIPTION
**Pull request by can-anyscale**
`tfrecords_datasource` has two APIs: `TFRecordDatasource` should be internal and `TFXReadOptions` should be public. To do so,  I splitted `tfrecords_datasource.py` into two parts.

Test:
- CI

### CODE REVIEW

File: doc/source/data/api/data_source.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.. _data-source-api:
+
+Global configuration
```
Comment (bveeramani) [inline doc/source/data/api/data_source.rst:3]: To minimize clutter in the sidebar and improve the discoverability of `TFXReadOptions`, could we co-locate this reference with `read_tfrecords` in the Input/Output page? I don't think we should place `TFXReadOptions` on a page by itself if it's only used with `read_tfrecords`

Comment (can-anyscale) [reply doc/source/data/api/data_source.rst:3]: totally

**Comment by can-anyscale** [comment]
@bveeramani's comments

**Review by bveeramani — APPROVED** [review]
Nice. Ty

**Pull request merged**